File: 0708/문자열 압축.py
# 1개 이상 단위로 문자열을 잘라 압축하여 표현한 문자열 중 가장 짧은 것의 길이
def compress(s, c):
    result = ''
    n = len(s)
    i = 0
    while i < n:
        part = s[i:i+c]
        repeated = 1
        while s[i+c:i+2*c] == part:
            repeated += 1
            i += c
        if repeated > 1:
            result += str(repeated) + part
        else:
            result += part
        i += c
    return len(result)

def solution(s):
    minLength = len(s)
    for i in range(1, len(s)//2+1):
        minLength = min(minLength, compress(s, i))
    return minLength

File: 0708/HyowonPark/문자열압축.js
function solution(s) {
     let minLength = s.length;
    
    for (let unit = 1; unit <= Math.floor(s.length / 2); unit++) {
        const groups = [];  // [[3, 패턴]]
        let prev = s.substring(0, unit);
        let count = 1;
        
        for (let i = unit; i < s.length; i += unit) {
            const current = s.substring(i, i + unit);
            
            if (prev === current) {
                count++;
            } else {
                groups.push([count, prev]);
                prev = current;
                count = 1;
            }
        }
        
        groups.push([count, prev]);
        
        const result = groups
            .map(([cnt, pattern]) => (cnt > 1 ? cnt : '') + pattern)
            .join('');
        
        minLength = Math.min(minLength, result.length);
    }
    
    return minLength;
}